airbyte-integrations/connectors/source-typeform/source_typeform/source.py

```diff
@@ -0,0 +1,229 @@
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+
+import urllib.parse as urlparse
+from abc import ABC, abstractmethod
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Tuple
+from urllib.parse import parse_qs
+
+import pendulum
+import requests
+from airbyte_cdk import AirbyteLogger
+from airbyte_cdk.models import SyncMode
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.http import HttpStream
+from airbyte_cdk.sources.streams.http.auth import TokenAuthenticator
+from pendulum.datetime import DateTime
+
+
+class TypeformStream(HttpStream, ABC):
+    url_base = "https://api.typeform.com"
+    limit: int = 200
+    date_format: str = "YYYY-MM-DDTHH:mm:ss[Z]"
+
+    def __init__(self, **kwargs: Mapping[str, Any]):
+        super().__init__(authenticator=kwargs["authenticator"])
+        self.config: Mapping[str, Any] = kwargs
+        self.start_date: DateTime = pendulum.from_format(kwargs["start_date"], self.date_format)
+
+        # changes page limit, this param is using for development and debugging
+        if kwargs.get("page_size"):
+            self.limit = kwargs.get("page_size")
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        return None
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        yield from response.json()["items"]
+
+
+class TrimForms(TypeformStream):
+    """
+    This stream is responsible for fetching list of from_id(s) which required to process data from Forms and Responses.
+    API doc: https://developer.typeform.com/create/reference/retrieve-forms/
+    """
+
+    primary_key = "id"
+    limit: int = 200
+
+    def path(
+        self,
+        stream_state: Mapping[str, Any] = None,
+        stream_slice: Mapping[str, Any] = None,
+        next_page_token: Mapping[str, Any] = None,
+    ) -> str:
+        return "/forms"
+
+    def next_page_token(self, response: requests.Response) -> Optional[int]:
+        page = self.get_current_page_token(response.url)
+        # stop pagination if current page equals to total pages
+        return None if response.json()["page_count"] <= page else page + 1
```

Comment:
Do we need to add additional check that page is not None, 

I am asking because there can be an error if we compare str or int with None

Reply:
ok, but in this case page can't be None